src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -45,74 +48,71 @@ protected internal override void Setup(FeatureConfigurationContext context)
+                var maxRetries = GetMaxRetries(context.Settings);
                 var retryPolicy = new FirstLevelRetryPolicy(maxRetries);
 
-                incomingStepSequence.Register("FirstLevelRetries", b => new FirstLevelRetriesBehavior(failureInfoStorage, retryPolicy), "Performs first level retries");
+                context.Pipeline.Register("FirstLevelRetries", b => new FirstLevelRetriesBehavior(failureInfoStorage, retryPolicy), "Performs first level retries");
             }
 
             RaiseLegacyNotifications(context);
         }
 
         static SecondLevelRetryPolicy GetDelayedRetryPolicy(ReadOnlySettings settings)
         {
-            var customRetryPolicy = settings.GetOrDefault<Func<IncomingMessage, TimeSpan>>("SecondLevelRetries.RetryPolicy");
-
-            if (customRetryPolicy != null)
+            Func<IncomingMessage, TimeSpan> customRetryPolicy;
+            if (settings.TryGet(SlrCustomPolicy, out customRetryPolicy))
             {
                 return new CustomSecondLevelRetryPolicy(customRetryPolicy);
             }
 
+            var numberOfRetries = settings.Get<int>(SlrNumberOfRetries);
+            var timeIncrease = settings.Get<TimeSpan>(SlrTimeIncrease);
+
             var retriesConfig = settings.GetConfigSection<SecondLevelRetriesConfig>();
             if (retriesConfig != null)
             {
-                return new DefaultSecondLevelRetryPolicy(retriesConfig.NumberOfRetries, retriesConfig.TimeIncrease);
+                numberOfRetries = retriesConfig.Enabled ? retriesConfig.NumberOfRetries : 0;
+                timeIncrease = retriesConfig.TimeIncrease;
             }
 
-            return new DefaultSecondLevelRetryPolicy(DefaultSecondLevelRetryPolicy.DefaultNumberOfRetries, DefaultSecondLevelRetryPolicy.DefaultTimeIncrease);
+            return new DefaultSecondLevelRetryPolicy(numberOfRetries, timeIncrease);
         }
 
         bool IsDelayedRetriesEnabled(ReadOnlySettings settings)
         {
-            if (!settings.Get<bool>(DelayedRetriesEnabled))
+            //Transactions must be enabled since SLR requires the transport to be able to rollback
+            if (settings.GetRequiredTransactionModeForReceives() == TransportTransactionMode.None)
             {
                 return false;
             }
 
-            //Transactions must be enabled since SLR requires the transport to be able to rollback
-            if (settings.GetRequiredTransactionModeForReceives() == TransportTransactionMode.None)
+            Func<IncomingMessage, TimeSpan> customPolicy;
+            if (settings.TryGet(SlrCustomPolicy, out customPolicy))
             {
-                return false;
+                return true;
             }
 
             var retriesConfig = settings.GetConfigSection<SecondLevelRetriesConfig>();
-
-            if (retriesConfig == null)
+            if (retriesConfig != null && retriesConfig.Enabled && retriesConfig.NumberOfRetries > 0)
```

Comment:
This change from `retriesConfig == null` to `retriesConfig != null` is interesting, because I'm not sure why this was returning true before.